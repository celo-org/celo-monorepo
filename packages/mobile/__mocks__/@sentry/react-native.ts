const Sentry = {
  config: () => ({ install: jest.fn() }),
  setTagsContext: jest.fn(),
  captureException: jest.fn(),
}

module.exports = Sentry
