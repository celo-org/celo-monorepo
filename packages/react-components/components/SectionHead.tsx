import colors from '@celo/react-components/styles/colors'
import { fontStyles } from '@celo/react-components/styles/fonts'
import * as React from 'react'
import { StyleSheet, Text, View } from 'react-native'

interface Props {
  text: string
  bubbleText?: string | null
}

export default class SectionHead extends React.PureComponent<Props> {
  render() {
    return (
      <View style={style.sectionHead}>
        <Text style={fontStyles.sectionLabel}>{this.props.text}</Text>
        {this.props.bubbleText && (
          <Text style={[fontStyles.notification, style.bubble]}>{this.props.bubbleText}</Text>
        )}
      </View>
    )
  }
}

const style = StyleSheet.create({
  sectionHead: {
    backgroundColor: colors.gray1,
    paddingHorizontal: 10,
    height: 36,
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  bubble: {
    textAlign: 'center',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-around',
    backgroundColor: colors.onboardingBlue,
    borderRadius: 10,
    padding: 1,
    margin: 5,
    height: 20,
    width: 20,
  },
})
