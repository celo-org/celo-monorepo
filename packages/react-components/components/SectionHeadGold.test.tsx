import SectionHead from '@celo/react-components/components/SectionHeadGold'
import * as React from 'react'
import 'react-native'
import * as renderer from 'react-test-renderer'

it('renders text', () => {
  const tree = renderer.create(<SectionHead text={'This is a Test'} />)
  expect(tree).toMatchSnapshot()
})
