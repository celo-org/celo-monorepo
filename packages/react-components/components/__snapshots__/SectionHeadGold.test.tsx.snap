// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders text 1`] = `
<View
  style={
    Object {
      "paddingBottom": 8,
      "paddingHorizontal": 16,
      "paddingTop": 20,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#2E3338",
        "fontFamily": "Jost-Medium",
        "fontSize": 22,
        "lineHeight": 28,
      }
    }
  >
    This is a Test
  </Text>
</View>
`;
